### ISSUE TITLE
`ExternalTexturesTest` still sometimes crashes on rendering the Android image

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#144365]
[Example output](https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20Engine%20Drone/2061528/overview):

![image](https://github.com/flutter/flutter/assets/168174/37c4efcc-db3a-454f-a33e-9c9cf80eefc8)

```txt
02-29 00:15:56.463  4679  5375 I Scenarios: Surface disconnected from ImageWriter
02-29 00:15:56.463  4679  5375 I Scenarios: java.lang.IllegalStateException: nativeDetachImage failed for image!!!
02-29 00:15:56.463  4679  5375 I Scenarios:     at android.media.ImageReader.nativeDetachImage(Native Method)
02-29 00:15:56.463  4679  5375 I Scenarios:     at android.media.ImageReader.detachImage(ImageReader.java:901)
02-29 00:15:56.463  4679  5375 I Scenarios:     at android.media.ImageWriter.queueInputImage(ImageWriter.java:512)
02-29 00:15:56.463  4679  5375 I Scenarios:     at dev.flutter.scenarios.ExternalTextureFlutterActivity$ImageSurfaceRenderer.onImageAvailable(ExternalTextureFlutterActivity.java:412)
02-29 00:15:56.463  4679  5375 I Scenarios:     at dev.flutter.scenarios.ExternalTextureFlutterActivity$ImageSurfaceRenderer.$r8$lambda$BBcQh_YQGJ8Oq2zMsGvTmZhrjIk(Unknown Source:0)
02-29 00:15:56.463  4679  5375 I Scenarios:     at dev.flutter.scenarios.ExternalTextureFlutterActivity$ImageSurfaceRenderer$$ExternalSyntheticLambda2.onImageAvailable(Unknown Source:2)
02-29 00:15:56.463  4679  5375 I Scenarios:     at android.media.ImageReader$1.run(ImageReader.java:947)
02-29 00:15:56.463  4679  5375 I Scenarios:     at android.os.Handler.handleCallback(Handler.java:958)
02-29 00:15:56.463  4679  5375 I Scenarios:     at android.os.Handler.dispatchMessage(Handler.java:99)
02-29 00:15:56.463  4679  5375 I Scenarios:     at android.os.Looper.loopOnce(Looper.java:205)
02-29 00:15:56.463  4679  5375 I Scenarios:     at android.os.Looper.loop(Looper.java:294)
02-29 00:15:56.463  4679  5375 I Scenarios:     at android.os.HandlerThread.run(HandlerThread.java:67)
02-29 00:15:56.469  4679  5375 V Scenarios: Output image
02-29 00:15:56.540   548  2292 I ImeTracker: dev.flutter.scenarios:e10aac05: onRequestHide at ORIGIN_SERVER_HIDE_INPUT reason HIDE_UNSPECIFIED_WINDOW
02-29 00:15:56.540   548  2292 I ImeTracker: dev.flutter.scenarios:e10aac05: onCancelled at PHASE_SERVER_SHOULD_HIDE
```

The code is [here](https://github.com/flutter/engine/blob/232217f39c3c64bcdb46a84cfb21d892e657d148/testing/scenario_app/android/app/src/main/java/dev/flutter/scenarios/ExternalTextureFlutterActivity.java#L346-L467), i.e. `ImageSurfaceRenderer` in `ExternalTextureFlutterActivity.java`. There is likely a bug in the implementation, we should look at other `ImageReader`-based code and/or ask @johnmccutchan for help.

**Comment by matanlurey** [comment]
Something to look at/solve next week.

**Comment by matanlurey** [comment]
@jonahwilliams might help look at this, so temporarily assigning to him. Please assign back if not the case.

**Comment by matanlurey** [comment]
@jonahwilliams @dnfield and I chatted informally - we think usefulness of having the Android rendered image next to the Flutter rendered image has outlived it's usefulness: it's both confusing to engine failure triage ("but I see one image, I guess it looks fine") _and_ means we need to keep Android-specific code branches that are not part of the actual engine, i.e. ImageReader/ImageWriter in the scenario activities, working.

Instead, I've proposed deleting the Android-specific code path to solve this problem.
